x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportUpdateDatafeedAction.java

```diff
@@ -86,13 +89,28 @@ protected void masterOperation(Task task, UpdateDatafeedAction.Request request,
 
         String datafeedId = request.getUpdate().getId();
 
-        CheckedConsumer<Boolean, Exception> updateConsumer = ok -> {
-            datafeedConfigProvider.updateDatefeedConfig(request.getUpdate().getId(), request.getUpdate(), headers,
-                    jobConfigProvider::validateDatafeedJob,
-                    ActionListener.wrap(
-                            updatedConfig -> listener.onResponse(new PutDatafeedAction.Response(updatedConfig)),
-                            listener::onFailure
-                    ));
+        CheckedConsumer<Boolean, Exception> updateConsumer = unused1 -> {
+            datafeedConfigProvider.getDatafeedConfig(
+                datafeedId,
+                ActionListener.wrap(
+                    datafeedConfigBuilder -> {
+                        String jobId = datafeedConfigBuilder.build().getJobId();
+                        JobDataDeleter jobDataDeleter = new JobDataDeleter(client, jobId);
+                        jobDataDeleter.deleteDatafeedTimingStats(
```

Comment:
I think the timing stats should only be deleted if the job ID is changed.  As it stands some small change like increasing the query delay slightly will reset the timing stats.